### DISCUSSION TITLE
Secure setup using Tomcat WAR

### DISCUSSION THREAD

**Discussion by bripakes** [rhuss/jolokia#759] [General]
I was able to setup the unsecured Jolokia agent (using Tomcat 10.1.30 with SSL) and have Metricbeat send the data to Elasticsearch. The process was very straightforward - thank you for making the WAR file!! But now I am trying to use the secure agent and it is not working. I have added a user with the appropriate role in my tomcat-users.xml, but it seems I need to "setup your servlet container to connect this role to the authentication". Can some provide more details on how to do this? The manual mentions a login-config and a security-constraint but I'm afraid I need a bit more information... Thank you, Brian

**Comment by bripakes** [comment]
Getting a little farther... It looks like I need to modify $TC/webapps/jolokia-agent-war-unsecured-2.1.1/WEB-INF/web.xml. I can uncomment the last section. What other changes do I need to make here? Last time I used the secure agent, I would get the basic auth prompt but it would not accept my credentials in $TC/conf/tomcat-users.xml (the one with the "jolokia" role).

**Comment by grgrzybek** [comment]
You shouldn't be forced to modify `web.xml`... Here's my user from `conf/tomcat-users.xml`:
```xml
<user username="jolokia" password="jolokia" roles="jolokia" />
```

and it should be everything that's needed to run the _secured_ WAR...

**Comment by bripakes** [comment]
Thank you @grgrzybek. I had 2 issues on my end that, once resolved, allowed the secure agent to work!
1) My passwords are encrypted in tomcat-users.xml. So I had neglected to encrypt the jolokia password.
2) I needed to add username and password entries in my metricbeat module.

**Reply by grgrzybek** [reply]
thanks for this summary ;)
